tests/unit/splash/test_splash_model.py

```diff
@@ -67,19 +67,29 @@ def test_splash_model_init(splash_core_constants, grid_benchmarks, var, flag):
 def splash_model(grid_benchmarks, splash_core_constants, calendar):
     """Create a SplashModel object for testing."""
 
+    from pyrealm.core.calendar import Calendar
     from pyrealm.splash.splash import SplashModel
 
     ds = grid_benchmarks[0].sel(time=calendar.dates)
 
-    splash = SplashModel(
-        lat=np.broadcast_to(ds.lat.data[None, :, None], ds.sf.data.shape),
-        elv=np.broadcast_to(ds.elev.data[None, :, :], ds.sf.data.shape),
-        dates=calendar,
-        sf=ds.sf.data,
-        tc=ds.tmp.data,
-        pn=ds.pre.data,
-        core_const=splash_core_constants,
-    )
+    for test_time_index_check in [True, False]:
+        if test_time_index_check:
+            dates = Calendar(calendar.dates[:-1])
+            context = pytest.raises(ValueError)
+        else:
+            dates = calendar
+            context = nullcontext()
```

Comment:
I don't think this is in the right place. 

Fixtures provide shared data and functionality to different tests and I don't think are intended to implement actual testing. Obviously you _can_ put an `assert` in here, but this fixture is created independently multiple times (it has "function" scope), so this test is repeated every time the fixture is used.

I'd also tend to implement the `test_time_index_check` using `pytest.mark.parameterise` rather than an internal loop, but that's more minor.

Reply:
I've put it into `test_splash_model_init`, using `dates` as an additional parameter now.

Reply:
Thanks - that looks better!